src/RA_Interface.h

```diff
@@ -40,81 +40,85 @@ enum EmulatorID
-    Arcade = 27,
-    VirtualBoy = 28,
-    MSX = 29,
-    C64 = 30,
-    ZX81 = 31,
-    //unused32 = 32,
-    SG1000 = 33,
-    VIC20 = 34,
-    Amiga = 35,
-    AmigaST = 36,
-    AmstradCPC = 37,
-    AppleII = 38,
-    Saturn = 39,
-    Dreamcast = 40,
-    PSP = 41,
-    CDi = 42,
-    ThreeDO = 43,
-    Colecovision = 44,
+    MegaDrive        = 1, //	DB
+    N64              = 2,
+    SNES             = 3,
+    GB               = 4,
+    GBA              = 5,
+    GBC              = 6,
+    NES              = 7,
+    PCEngine         = 8,
+    SegaCD           = 9,
+    Sega32X          = 10,
+    MasterSystem     = 11,
+    PlayStation      = 12,
+    Lynx             = 13,
+    NeoGeoPocket     = 14,
+    GameGear         = 15,
+    GameCube         = 16,
+    Jaguar           = 17,
+    DS               = 18,
+    WII              = 19,
+    WIIU             = 20,
+    PlayStation2     = 21,
+    Xbox             = 22,
+    Events           = 23, // not an actual console
+    XboxOne          = 24,
+    Atari2600        = 25,
+    MSDOS            = 26,
+    Arcade           = 27,
+    VirtualBoy       = 28,
+    MSX              = 29,
+    C64              = 30,
+    ZX81             = 31,
+    // unused32 = 32,
+    SG1000        = 33,
+    VIC20         = 34,
+    Amiga         = 35,
+    AmigaST       = 36,
+    AmstradCPC    = 37,
+    AppleII       = 38,
+    Saturn        = 39,
+    Dreamcast     = 40,
+    PSP           = 41,
+    CDi           = 42,
+    ThreeDO       = 43,
+    Colecovision  = 44,
     Intellivision = 45,
-    Vectrex = 46,
-    PC8800 = 47,
-    PC9800 = 48,
-    PCFX = 49,
-    Atari5200 = 50,
-    Atari7800 = 51,
-    X68K = 52,
-    WonderSwan = 53,
+    Vectrex       = 46,
+    PC8800        = 47,
+    PC9800        = 48,
+    PCFX          = 49,
+    Atari5200     = 50,
+    Atari7800     = 51,
+    X68K          = 52,
+    WonderSwan    = 53,
 
     NumConsoleIDs
 };
 
-
-extern bool(*_RA_GameIsActive)();
-extern void(*_RA_CauseUnpause)();
-extern void(*_RA_CausePause)();
-extern void(*_RA_RebuildMenu)();
-extern void(*_RA_GetEstimatedGameTitle)(char* sNameOut);
-extern void(*_RA_ResetEmulation)();
-extern void(*_RA_LoadROM)(const char* sFullPath);
-
-//	Shared funcs, should be implemented by emulator.
+extern bool (*_RA_GameIsActive)();
+extern void (*_RA_CauseUnpause)();
+extern void (*_RA_CausePause)();
+extern void (*_RA_RebuildMenu)();
+extern void (*_RA_GetEstimatedGameTitle)(char* sNameOut);
+extern void (*_RA_ResetEmulation)();
+extern void (*_RA_LoadROM)(const char* sFullPath);
+
+// Shared funcs, should be implemented by emulator.
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
 extern bool RA_GameIsActive();
 extern void RA_CauseUnpause();
 extern void RA_CausePause();
 extern void RA_RebuildMenu();
 extern void RA_GetEstimatedGameTitle(char* sNameOut);
 extern void RA_ResetEmulation();
 extern void RA_LoadROM(const char* sFullPath);
-
+#ifdef __cplusplus
+}
+#endif // __cplusplus
```

Comment:
I put this around an `extern "C"` block because the analyzer was telling me to mark them `noexcept`, but wasn't sure if an emulator supports at least C++11.